Tests/MongoSwiftTests/CodecTests.swift

```diff
@@ -716,11 +716,15 @@ final class CodecTests: MongoSwiftTestCase {
                                   from: wrappedMinKey.canonicalExtendedJSON).x.value as? MinKey).to(equal(minKey))
     }
 
-    struct OptionalAnyBSONWrapper: Codable {
-        let val: AnyBSONValue?
+    struct OptionalAnyBSONWrapper: Codable, Equatable {
```

Comment:
maybe we should rename this to get rid of the `Optional`...

Reply:
I agree that the "Optional" name prefix doesn't make sense, but is there even a point to this class? I imagine we could just use `AnyBSONValue` directly.

Reply:
you can, i think my intent here was just to check that encoding and decoding worked as expected when the type wasn't directly encoded/decoded and was optional. which one could argue is something we should just rely on codable to get right and not test ourselves....
however, now that this is a non optional, this test seems redundant with `testAnyBSONValueIsBSONCodable`, which uses an identical `AnyBSONStruct`

Reply:
Oh right, good point. Should I just cut this part of the test out?

Reply:
ya, unless if there's anything we can find here that the previous test isn't checking, we can scrap it

Reply:
Done.

I migrated over the testing code for `NSNull` in `testOptionalAnyBSONValue()` as well as the test for array that is not all `BSONValue`. That test actually calls `encode()` from the `BSONValue` protocol... which seems a little odd if its not related to `BSONEncoder`/`BSONDecoder`, but oh well.